### PR TITLE
Show keymap descriptions for sub-keymaps (Fix #306)

### LINKED ISSUE #306: Verbose indicator: show keymap name/description for prefix keymaps
Afaik the name is available for keymaps set via fset.

### PR DESCRIPTION
**Pull request by minad**

### CODE REVIEW

**Comment by minad** [comment]
@oantolin If you dislike this approach, we could also use mapatoms and cache the variable name and documentation. I am not entirely sure. Or one has to support all different variants of how these prefix keymaps can be bound. Someone could also use `(fset 'map (make-sparse-map))` without a separate bound variable.

File: embark.el [hunk 1]
```diff
@@ -838,8 +842,10 @@ first line of the documentation string; otherwise use the word
   (concat ; fresh copy, so we can freely add text properties
    (cond
     ((stringp (car-safe cmd)) (car cmd))
+    ((keymapp cmd)
+     (propertize (if (symbolp cmd) (format "+%s" cmd) "<keymap>")
```
Comment (oantolin) [inline embark.el:846]: What is the plus sign for? Is that a which-key convention?

File: embark.el [hunk 2]
```diff
@@ -883,7 +894,7 @@ If NESTED is non-nil subkeymaps are not flattened."
                            (eq cmd #'embark-keymap-help))
                    collect (list name
                                  (cond
-                                  ((keymapp cmd) 'keymap)
+                                  ((and (not (symbolp cmd)))(keymapp cmd) 'keymap)
```
Comment (oantolin) [inline embark.el:897]: I'm guessing you wanted the keymapp inside the and.

File: embark.el (hunk 1 above)
Comment (minad) [reply embark.el:846]: Yes, which-key does this. We could add a defcustom to format this?

File: embark.el (hunk 2 above)
Comment (minad) [reply embark.el]: Fixed.

File: embark.el (hunk 1 above)
Comment (oantolin) [reply embark.el:846]: I guess the plus is OK, which-key is probably the most recognized comencemos for this. I prefer not to add yet another defcustom unless someone really wants it.

**Comment by oantolin** [comment]
OK, Emacs does seem to use this funny `fset` pattern for some keymaps, so I guess it's not just crazy which-key users that do that. 😛

**Comment by minad** [comment]
It is still crazy. But at least you shouldn't feel bad now to put functions into keymaps ;)

**Pull request merged**